receiver/k8sclusterreceiver/metricsstore.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sclusterreceiver
+
+import (
+	"sync"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/utils"
+)
+
+// metricsStore keeps track of the metrics being pushed along the pipeline
+// every interval. Since Kubernetes events that generate these metrics are
+// aperiodic, the values in this cache will be pushed along the pipeline
+// until the next Kubernetes event pertaining to an object.
+type metricsStore struct {
+	sync.RWMutex
+	metricsCache map[types.UID][]consumerdata.MetricsData
+}
+
+// This probably wouldn't be required once the new OTLP ResourceMetrics
+// struct is made available.
+type resourceMetrics struct {
+	resource *resourcepb.Resource
+	metrics  []*metricspb.Metric
+}
+
+// updates metricsStore with latest metrics.
+func (ms *metricsStore) update(obj interface{}, rms []*resourceMetrics) error {
```

Comment:
Does `obj` need to be declared as weakly typed `interface{}`? It is unconditionally converted to `runtime.Object` below, so maybe it is possible to use that type in parameter declaration as well?

Reply:
Yes, that's a good point. Updated this method to accept `runtime.Object`.